config/src/main/java/org/springframework/security/config/annotation/web/configuration/HttpSecurityConfiguration.java

```diff
@@ -131,9 +133,14 @@ HttpSecurity httpSecurity() throws Exception {
 	}
 
 	private void applyCorsIfAvailable(HttpSecurity http) throws Exception {
-		String[] beanNames = this.context.getBeanNamesForType(CorsConfigurationSource.class);
-		if (beanNames.length == 1) {
-			http.cors(withDefaults());
+		Map<String, CorsConfigurationSource> corsConfigurationSources = this.context
+			.getBeansOfType(CorsConfigurationSource.class);
+
+		for (CorsConfigurationSource source : corsConfigurationSources.values()) {
+			if (source instanceof UrlBasedCorsConfigurationSource) {
+				http.cors(withDefaults());
```

Comment:
I wonder if we should check if the instance is `UrlBasedCorsConfigurationSource` and if the bean name is `corsConfigurationSource`, since this is the [bean name used by the `CorsConfigurer`](https://github.com/spring-projects/spring-security/blob/main/config/src/main/java/org/springframework/security/config/annotation/web/configurers/CorsConfigurer.java#L89), to avoid picking up the wrong `CorsConfigurationSource`.

Reply:
I have updated the code to check if there are any beans of type UrlBasedCorsConfigurationSource using getBeanNamesForType. If such beans exist, CORS configuration is applied.

If this is not what you intended, please feel free to provide further feedback.